File: frontend/src/services/AirportService.ts
import { apiClient } from "./api";
import { Airport, ApiResponse, PaginatedResponse } from "../models";

export class AirportService {
  private readonly baseUrl = "/airports";

  async getAllAirports(): Promise<ApiResponse<Airport[]>> {
    return apiClient.get(this.baseUrl);
  }

  async getAllAirportsPaged(
    page: number = 0,
    size: number = 10
  ): Promise<ApiResponse<PaginatedResponse<Airport>>> {
    return apiClient.get(`${this.baseUrl}?page=${page}&size=${size}`);
  }

  async getAirportById(id: number): Promise<ApiResponse<Airport>> {
    return apiClient.get(`${this.baseUrl}/${id}`);
  }

  async getAirportsByCity(cityName: string): Promise<ApiResponse<Airport[]>> {
    return apiClient.get(`${this.baseUrl}/city/${cityName}`);
  }

  async getAirportsByCountry(
    countryName: string
  ): Promise<ApiResponse<Airport[]>> {
    return apiClient.get(`${this.baseUrl}/country/${countryName}`);
  }

  async searchAirportsByName(name: string): Promise<ApiResponse<Airport[]>> {
    return apiClient.get(`${this.baseUrl}/search/${name}`);
  }

  async createAirport(
    airport: Omit<Airport, "airportId">
  ): Promise<ApiResponse<Airport>> {
    return apiClient.post(this.baseUrl, airport);
  }

  async updateAirport(
    id: number,
    airport: Partial<Airport>
  ): Promise<ApiResponse<Airport>> {
    return apiClient.put(`${this.baseUrl}/${id}`, airport);
  }

  async deleteAirport(id: number): Promise<ApiResponse<void>> {
    return apiClient.delete(`${this.baseUrl}/${id}`);
  }
}

export const airportService = new AirportService();

File: backend/src/main/java/com/flightmanagement/service/AirportService.java
package com.flightmanagement.service;

import com.flightmanagement.dto.AirportDto;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

import java.util.List;

public interface AirportService {
    
    List<AirportDto> getAllAirports();

    Page<AirportDto> getAllAirportsPaged(Pageable pageable);
    
    AirportDto getAirportById(Integer id);
    
    AirportDto createAirport(AirportDto airportDto);
    
    AirportDto updateAirport(Integer id, AirportDto airportDto);
    
    void deleteAirport(Integer id);
    
    List<AirportDto> getAirportsByCity(String cityName);
    
    List<AirportDto> getAirportsByCountry(String countryName);
    
    List<AirportDto> searchAirportsByName(String airportName);
}
